src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/mapper/HaFlowMapper.java

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2023 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowmonitoring.mapper;
+
+import static java.lang.String.format;
+
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.HaFlow;
+import org.openkilda.model.HaFlowPath;
+import org.openkilda.model.HaSubFlow;
+import org.openkilda.model.PathSegment;
+import org.openkilda.wfm.topology.flowmonitoring.model.FlowState;
+import org.openkilda.wfm.topology.flowmonitoring.model.Link;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Mapper
+public class HaFlowMapper {
+    public static final HaFlowMapper INSTANCE = Mappers.getMapper(HaFlowMapper.class);
+
+    /**
+     * Convert HA-Sub-Flow path representation.
+     */
+    public FlowState toFlowStates(HaSubFlow haSubFlow) {
+        HaFlow haFlow = haSubFlow.getHaFlow();
+
+        if (haFlow != null) {
+            FlowState flowState = new FlowState();
+            flowState.setForwardPath(toLinks(getPathSegments(haFlow.getForwardPath(), haSubFlow.getHaSubFlowId())));
+            flowState.setReversePath(toLinks(getPathSegments(haFlow.getReversePath(), haSubFlow.getHaSubFlowId())));
+            return flowState;
+        }
+        throw new IllegalStateException(format("HA-Flow for HA-Sub-Flow %s is not found.", haSubFlow.getHaFlowId()));
```

Comment:
If this exception will be thrown during cache initialization we will get log `Error on flow cache initialization: %s"`, empty cache and topology which is not active. You already handle it in `initCache` by checking that flow is incomplete. Please handle it in other palaces too, just check if HaFlow is null and log it (or catch IllegalArgumentException and handle in)